aspnetcore/release-notes/aspnetcore-7.0.md

```diff
@@ -209,6 +209,309 @@ SignalR hub methods now support injecting services through dependency injection
 
 Hub constructors can accept services from DI as parameters, which can be stored in properties on the class for use in a hub method. For more information, see [Inject services into a hub](xref:signalr/hubs?view=aspnetcore-7.0&preserve-view=true#inject-services-into-a-hub)
 
+## Blazor
+
+### Handle location changing events
+
+In .NET 7, Blazor supports location changing events. This allows you to warn users about unsaved work or to perform related actions when the user performs a page navigation.
+
+For more information, see the following sections of the *Routing and navigation* article:
+
+* [Navigation options](xref:blazor/fundamentals/routing?view=aspnetcore-7.0#navigation-options)
+* [Handle/prevent location changes](xref:blazor/fundamentals/routing?view=aspnetcore-7.0#handleprevent-location-changes)
+
+### Calling `InvokeAsync(StateHasChanged)` causes page to fallback to the default culture
+
+If a Razor component defines an event that's triggered from a background thread, the component might be required to capture and restore the execution context (`ExecutionContext`) at the time the handler is registered.
+
+For more information, see the following resources:
+
+* [Calling `InvokeAsync(StateHasChanged)` causes page to fallback to default culture (dotnet/aspnetcore #28521)](https://github.com/dotnet/aspnetcore/issues/28521)
+* <xref:blazor/components/index?view=aspnetcore-7.0#invoke-component-methods-externally-to-update-state>
+
+### Empty Blazor project templates
+
+Blazor has two new project templates for starting from a blank slate. The new **Blazor Server App Empty** and **Blazor WebAssembly App Empty** project templates are just like their non-empty counterparts but without demonstration code. These empty templates only include a basic home page, and we've removed Bootstrap so that you can start with a different CSS framework.
+
+For more information, see the following articles:
+
+* <xref:blazor/tooling?view=aspnetcore-7.0>
+* <xref:blazor/project-structure?view=aspnetcore-7.0>
+
+### Blazor custom elements no longer experimental
+
+The [`Microsoft.AspNetCore.Components.CustomElements`](https://www.nuget.org/packages/microsoft.aspnetcore.components.customelements) package for building [standards based custom elements](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-elements) is no longer experimental.
+
+For more information, see <xref:blazor/components/index?view=aspnetcore-7.0#blazor-custom-elements>.
+
+<!--
+
+    IN PROGRESS
+
+### Improve prerendering and authentication experience in Blazor
+
+[27592](https://github.com/dotnet/aspnetcore/issues/27592)
+
+-->
+
+### Bind get/set/after modifiers
+
+New modifiers, `@bind:get`/`@bind:set`/`@bind:after`, allow you to customize how Blazor's `@bind` attribute directive reads and writes bound values.
+
+<!--
```

Comment:
@guardrex can you bring some content from [the ASP.NET Core Preview 7 release blog post](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-7-preview-7/#blazor-data-binding-get-set-after-modifiers)? Or at least link it as it has some good content about new bind methods:

Reply:
Ah! Thanks for calling that out. I'm not sure why the bit at the end on `@bind:get`/`@bind:set` escaped my 👀 earlier. 

I'll steal Dan's content to beef up this *What's New* section, and I'll flesh out the `get`/`set` piece in the *Data binding* topic with a full explanation and example.

I'm going to take care of it first thing Monday morning.